src/main/java/de/cas/mse/exercise/diamond/Diamond.java

```diff
@@ -3,29 +3,38 @@
 import java.util.Arrays;
 
 public class Diamond {
+    public static void main(String[] args) {
+        System.out.println(new Diamond().generateDiamond(5));
+    }
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
+	public String generateDiamond(int height) {
+		if (!isValidDiamondShape(height)) {
 			return null;
+        }
+        String middleLine = new String(makeLine(height, height));
+        StringBuilder diamond = new StringBuilder(middleLine);
+        // iteratively expand diamond towards top and bottom
```

Comment:
Instead of a line comment, could the insert and append methods be extracted so the code documents itself?